File: Fragment/Fragment_demo2/app/src/main/java/com/example/fragment_demo2/FirstFragment.java
package com.example.fragment_demo2;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.Toast;

public class FirstFragment extends Fragment {

    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    // TODO: Rename and change types of parameters
    private String mParam1;
    private String mParam2;

    public FirstFragment() {
        // Required empty public constructor
    }

    public static FirstFragment newInstance(String param1, String param2) {
        FirstFragment fragment = new FirstFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    // phương thức khởi tạo các giao diện và các giá trị cho chúng
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_first, container, false);

        Button btnItem1 = view.findViewById(R.id.btnItem1);
        btnItem1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(getActivity(), "button 1 onclick  ", Toast.LENGTH_SHORT).show();
            }
        });

        Button btnItem2 = view.findViewById(R.id.btnItem2);
        btnItem2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // getActivity : sẽ tự động gọi lấy Activity đang chứa fragment này
                Toast.makeText(getActivity(), "button 2  onlick ", Toast.LENGTH_SHORT).show();
            }
        });
        // Inflate the layout for this fragment
        return view;
    }
}

File: Fragment/Fragment_Api_demo3/app/src/main/java/com/example/fragment_api_demo3/FirstFragment.kt
package com.example.fragment_api_demo3

import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.setFragmentResultListener
import androidx.navigation.fragment.findNavController
import com.example.fragment_api_demo3.databinding.FragmentFirstBinding

class FirstFragment : Fragment() {
    private lateinit var bindding:FragmentFirstBinding

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        bindding = FragmentFirstBinding.inflate(inflater, container, false)
        navigateToSecoundScreen()
        setFragmnetListener()
        return bindding.root

    }

    private fun setFragmnetListener() {
        setFragmentResultListener(AppConstants.REQUEST_KEY){requestKey, bundle ->  
            val result = bundle.getString(AppConstants.BUNDLE_KEY)
            bindding.txtResult.text = result
        }
    }

    private fun navigateToSecoundScreen() {
        bindding.btnResult.setOnClickListener {
            findNavController().navigate(FirstFragmentDirections.actionFirstFragmentToSecoundFragment())
        }
    }
}